### PR TITLE
Add GKE Gateway v1.1 conformance report

### PR DESCRIPTION
**Pull request by wstcliyu**
**What type of PR is this?**

/area conformance

**What this PR does / why we need it**:
Add conformance report for GKE Gateway v1.1

**Which issue(s) this PR fixes**:

N/A

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - HTTPRouteBackendTimeout
+    - HTTPRouteMethodMatching
+    - HTTPRouteParentRefPort
+    - HTTPRoutePathRewrite
+    - HTTPRoutePortRedirect
+    - HTTPRouteQueryParamMatching
+    - HTTPRouteRequestMultipleMirrors
+    - HTTPRouteRequestTimeout
+  name: GATEWAY-HTTP
+  summary: Core tests partially succeeded with 1 test skips. Extended tests succeeded.
```
Comment (robscott) [inline conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml:49]: Nit: Recommend adding trailing newline here.

File: conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+gatewayAPIVersion: v1.1.0
+implementation:
+  contact:
+  - gke-gateway-dev@google.com
+  organization: GKE
+  project: gke-gateway
+  url: https://cloud.google.com/kubernetes-engine/docs/concepts/gateway-api
+  version: 1.30.3-gke.1211000
+kind: ConformanceReport
+mode: default
```
Comment (robscott) [inline conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml:13]: We probably eventually want to have one of these reports for each GatewayClass, so I'd recommend using mode to represent the GatewayClass used to generate this report.

**Review by robscott — COMMENTED** [review]
Thanks @wstcliyu!

File: conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+apiVersion: gateway.networking.k8s.io/v1alpha1
+date: "2024-07-28T18:12:57Z"
+gatewayAPIChannel: standard
+gatewayAPIVersion: v1.1.0
+implementation:
+  contact:
+  - gke-gateway-dev@google.com
```
[the remark this answers is not in the document]
Comment (wstcliyu) [reply conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml]: Done.

File: conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml (hunk 1 above)
Comment (wstcliyu) [reply conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml]: Done.

File: conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml (hunk 2 above)
Comment (wstcliyu) [reply conformance/reports/v1.1.0/gke-gateway/standard-1.30.3-default-report.yaml]: Done.

**Comment by robscott** [comment]
Thanks @wstcliyu!

/lgtm

Will defer to someone from a different company to approve.

File: conformance/reports/v1.1.0/gke-gateway/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# GKE (Google Kubernetes Engine) Gateway
```
Comment (robscott) [inline conformance/reports/v1.1.0/gke-gateway/README.md:1]: Actually forgot one step - do you mind updating GKE's entry in implementations.md to show the partial conformance status?

File: site-src/implementations.md [hunk 5]
```diff
@@ -250,6 +250,8 @@ Gloo Gateway 2.0 brings the full power and community support of Gateway API to i
 
 ### Google Kubernetes Engine
 
+[![Conformance](https://img.shields.io/badge/Gateway_API_Conformance_v1.1.0-Google_Kubernetes_Engine-green)](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/reports/v1.1.0/gke-gateway)
```
Comment (robscott) [inline site-src/implementations.md:253]: Nit: Looks slightly different for partial conformance. I think this should work:

```suggestion
[![Conformance](https://img.shields.io/badge/Gateway_API_Partial_Conformance_v1.1.0-Google_Kubernetes_Engine-orange)](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/reports/v1.1.0/gke-gateway)
```

If not, you can take a look at [this line](https://github.com/kubernetes-sigs/gateway-api/blob/3be634952484373c35add0408543fe404c6328a8/site-src/implementations.md#L139
) for reference.

Comment (wstcliyu) [reply site-src/implementations.md]: Done. Thank you. It works.

File: site-src/implementations.md [hunk 6]
```diff
@@ -260,6 +262,9 @@ advanced routing, multi-cluster load balancing and more. See the docs to deploy
 [private or public Gateways][gke-gateway-deploy] and also [multi-cluster
 Gateways][gke-multi-cluster-gateway].
 
+The GKE Gateway controller passes all the core Gateway API conformance tests
```
Comment (mlavacca) [inline site-src/implementations.md:265]:
```suggestion
The GKE Gateway controller passes all the core Gateway API conformance tests for the GATEWAY_HTTP conformance profile.
```

Feel free to rephrase it as you prefer, but we should make clear that  we are talking about the GATEWAY-HTTP conformance profile.

**Review by mlavacca — COMMENTED** [review]
Thanks, @wstcliyu!

File: site-src/implementations.md (hunk 6 above)
Comment (wstcliyu) [reply site-src/implementations.md]: Done. Thank you.

**Review by mlavacca — COMMENTED** [review]
/approve

**Comment by robscott** [comment]
Thanks @wstcliyu!

/lgtm

**Pull request merged**